src/rules/valid-expect.ts

```diff
@@ -0,0 +1,297 @@
+  | 'asyncMustBeAwaited'
+  | 'promisesWithAsyncAssertionsMustBeAwaited';
+
+export default createRule<[{ alwaysAwait?: boolean }], MessageIds>({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Enforce valid `expect()` usage',
+      recommended: 'error',
+    },
+    messages: {
+      multipleArgs: 'More than one argument was passed to expect().',
+      noArgs: 'No arguments were passed to expect().',
+      noAssertions: 'No assertion was called on expect().',
+      invalidProperty:
+        '"{{ propertyName }}" is not a valid property of expect.',
+      propertyWithoutMatcher: '"{{ propertyName }}" needs to call a matcher.',
+      matcherOnPropertyNotCalled: '"{{ propertyName }}" was not called.',
+      asyncMustBeAwaited: 'Async assertions must be awaited{{ orReturned }}.',
+      promisesWithAsyncAssertionsMustBeAwaited:
+        'Promises which return async assertions must be awaited{{ orReturned }}.',
+    },
+    type: 'suggestion',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          alwaysAwait: {
+            type: 'boolean',
+            default: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [{ alwaysAwait: false }],
+  create(context, [{ alwaysAwait }]) {
+    // Context state
+    const arrayExceptions = new Set<string>();
+
+    const pushPromiseArrayException = (loc: TSESTree.SourceLocation) =>
+      arrayExceptions.add(promiseArrayExceptionKey(loc));
+
+    /**
+     * Promise method that accepts an array of promises,
+     * (eg. Promise.all), will throw warnings for the each
+     * unawaited or non-returned promise. To avoid throwing
+     * multiple warnings, we check if there is a warning in
+     * the given location.
+     */
+    const promiseArrayExceptionExists = (loc: TSESTree.SourceLocation) =>
+      arrayExceptions.has(promiseArrayExceptionKey(loc));
+
+    return {
+      CallExpression(node) {
+        if (!isExpectCall(node)) {
+          return;
+        }
+
+        const { expect, modifier, matcher } = parseExpectCall(node);
+
+        if (expect.arguments.length > 1) {
+          const secondArgumentLocStart = expect.arguments[1].loc.start;
+          const lastArgumentLocEnd =
+            expect.arguments[node.arguments.length - 1].loc.end;
+
+          context.report({
+            loc: {
+              end: {
+                column: lastArgumentLocEnd.column - 1,
+                line: lastArgumentLocEnd.line,
+              },
+              start: secondArgumentLocStart,
+            },
+            messageId: 'multipleArgs',
+            node,
+          });
+        } else if (expect.arguments.length === 0) {
+          const expectLength = getAccessorValue(expect.callee).length;
+          context.report({
+            loc: {
+              end: {
+                column: node.loc.start.column + expectLength + 1,
```

Comment:
why is this `start` and not `end`?

Reply:
b/c that's how it was originally :P

I tried to maintain the original logic as possible as much as possible, to reduce noise.
Feel free to open an issue, and assign it to me 😉

But yeah looking at this it should be `end`.